### ISSUE TITLE
[Bug]: `/health` with unpriv user API key gives a very confusing response

### ISSUE THREAD

**Issue by Manouchehri** [BerriAI/litellm#3972]
### What happened?

While trying to verifying #3957, I discovered the health endpoint seems completely broken for me now.

```sh
curl -X 'GET' \
  'https://example.com/health' \
  -H 'accept: application/json' \
  -H 'Authorization: Bearer REMOVED'
```

It's oddly not even triggering Sentry, which is very surprising.

```json
{
  "error": {
    "message": "Authentication Error, isinstance() arg 2 must be a type, a tuple of types, or a union",
    "type": "auth_error",
    "param": "None",
    "code": 401
  }
}
```

This was tested with v1.39.6 and v1.40.0.

### Relevant log output

```shell
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/litellm/proxy/proxy_server.py", line 1361, in user_api_key_auth
    if user_id_information is not None and isinstance("
                                           ^^^^^^^^^^^
```

### Twitter / LinkedIn details

https://www.linkedin.com/in/davidmanouchehri/

**Comment by Manouchehri** [comment]
Oh, my bad, this is because only the master key works for health checks. Hmm, I guess we could do a better error if possible?

**Comment by ishaan-jaff** [comment]
` if user_id_information is not None and isinstance("` this error message is fixed on 1.40.2
